modules/dbparser/radix.c

```diff
@@ -532,8 +532,13 @@ r_parser_float(gchar *str, gint *len, const gchar *param, gpointer state, RParse
   if (str[*len] == '-')
     (*len)++;
 
-  while (g_ascii_isdigit(str[*len]) || (!dot && str[*len] == '.' && (dot = TRUE)))
-    (*len)++;
+  while (g_ascii_isdigit(str[*len]) || (!dot && str[*len] == '.'))
+    {
+      if (str[*len] == '.')
+        dot = TRUE;
+
+      (*len)++;
+    }
```

Comment:
If we are at readability, I would suggest something like this:
```c
void read_digists(gchar *str, gint *len)
{
   while (str[*len] && g_ascii_isdigit(str[*len])) (*len)++;
}
```

```c
read_digits(str, len);
if (str[*len} && str[*len] == '.')
{
    read_digits(str, len);
}
```

Note: this also fixes that `.` are not valid number, but not solves `.12` (the later can be argued to be "valid").

Reply:
Done, thanks.